### PR TITLE
Help commands are hidden for non-authorized users

### LINKED ISSUE #8: Help command should hide inaccessible commands
@peterkos suggested that users that call `$help` should not be able to see commands that they do not have permission to run. For example, a regular user should not see help info for `$ready`, and a mentor should not see help info for `$maketopic`.

### PR DESCRIPTION
**Pull request by skyegallup**
Closes #8

### CODE REVIEW

**Review by Madrugaur — CHANGES_REQUESTED** [review]
Just a thought: Maybe it would be a good idea to reference the Role that a Topic has inside the Topic class as local variable. I.e 
public Topic {
    private final Role role;
etc..
} 
Something like this would cut down on the amount of searching you would need to do.

File: src/main/java/listeners/MainEventListener.java [hunk 1]
```diff
@@ -117,10 +128,16 @@ private void help(Member member, TextChannel channel, Server server, String[] ar
         embedBuilder.addField("$queue <topic>", "Add yourself to a queue.", false);
         embedBuilder.addField("$showqueue <topic>", "Show the people currently in queue.", false);
         embedBuilder.addField("$showtopics", "List all topics.", false);
-        embedBuilder.addField("$ready <topic> (mentor only)", "Retrieve the next person from the queue.", false);
-        embedBuilder.addField("$clear <topic> (mentor only)", "Clear the specified queue.", false);
-        embedBuilder.addField("$maketopic <name> (admin only)", "Create a new topic.", false);
-        embedBuilder.addField("$deletetopic <name> (admin only)", "Delete a topic.", false);
+
+        if (isMentor(member)) {
+            embedBuilder.addField("$ready <topic> (mentor only)", "Retrieve the next person from the queue.", false);
```
Comment (peterkos) [inline src/main/java/listeners/MainEventListener.java:131]: I like how the role is specified in the command name!

**Review by peterkos — APPROVED** [review]
LGTM, also, any plans to add docs for some methods? I'll admit, i'm a bit spoiled by authbot 😛

**Comment by skyegallup** [comment]
@peterkos I'll definitely be adding more documentation to the code soon

**Comment by skyegallup** [comment]
I replaced the `ArrayList<Topic>` in the Server class with a `HashMap<String, Topic>` with the topic name as a key. This should make searching for a topic by name a lot faster (O(n) -> O(1)). I also stored the Role for a Topic inside the Topic object itself, as @Madrugaur suggested.

**Pull request merged**